src/main/kotlin/traffic_simulation/Vehicle.kt

```diff
@@ -27,6 +28,10 @@ class Vehicle(val id: Int, val wannaDriveInHours: MutableList<Int>) {
         if (!wannaDriveInHours.contains(timestep)) {
             delay -= 1
         }
+
+        if (delay <= 0 && wannaDriveInHours.contains(timestep)){
```

Comment:
Ah, ok, wir haben ein unterschiedlichen Verständnis davon, was genau festgehalten werden soll. Damn it, wer hätte gedacht, dass das so unterschiedlich sein kann. :smiley:

Wenn du in einem richtigen Projekt wärst, müsstest du dir jetzt überlegen, was die echten Anforderungen sind. Wir können es uns einfacher machen:

* in einer Liste steht, wann ein Fahrzeug gefahren ist
* in der anderen Liste steht, wann es nicht fahren konnte

Ergibt auch Sinn, oder, und ist relativ einfach.